Index: core/src/main/scala/org/apache/spark/SparkEnv.scala
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/main/scala/org/apache/spark/SparkEnv.scala b/core/src/main/scala/org/apache/spark/SparkEnv.scala
--- a/core/src/main/scala/org/apache/spark/SparkEnv.scala	(revision de351e30a90dd988b133b3d00fa6218bfcaba8b8)
+++ b/core/src/main/scala/org/apache/spark/SparkEnv.scala	(revision 82632949d9bc33af45d08703053afeeae0f92d92)
@@ -345,7 +345,9 @@
       conf.get(BLOCK_MANAGER_PORT)
     }
 
-    val externalShuffleClient = if (conf.get(config.SHUFFLE_SERVICE_ENABLED)) {
+    val isOrs2Shuffle = conf.get("spark.shuffle.manager", "sort")
+      .startsWith("org.apache.spark.shuffle.Ors")
+    val externalShuffleClient = if (conf.get(config.SHUFFLE_SERVICE_ENABLED) && !isOrs2Shuffle) {
       val transConf = SparkTransportConf.fromSparkConf(conf, "shuffle", numUsableCores)
       Some(new ExternalBlockStoreClient(transConf, securityManager,
         securityManager.isAuthenticationEnabled(), conf.get(config.SHUFFLE_REGISTRATION_TIMEOUT)))
